src/webgl/p5.RendererGL.Retained.js

```diff
@@ -43,127 +56,24 @@ p5.RendererGL.prototype.createBuffers = function(gId, obj) {
   //initialize the gl buffers for our geom groups
   this._initBufferDefaults(gId);
 
-  this.gHash[gId].numberOfItems = obj.faces.length * 3;
-  this.gHash[gId].lineVertexCount = obj.lineVertices.length;
-
-  this._useColorShader();
-
-  this._bindBuffer(
-    this.gHash[gId].lineVertexBuffer,
-    gl.ARRAY_BUFFER,
-    this._flatten(obj.lineVertices),
-    Float32Array,
-    gl.STATIC_DRAW
-  );
-
-  this.curStrokeShader.enableAttrib(
-    this.curStrokeShader.attributes.aPosition.location,
-    3,
-    gl.FLOAT,
-    false,
-    0,
-    0
-  );
-
-  this._bindBuffer(
-    this.gHash[gId].lineNormalBuffer,
-    gl.ARRAY_BUFFER,
-    this._flatten(obj.lineNormals),
-    Float32Array,
-    gl.STATIC_DRAW
-  );
-
-  this.curStrokeShader.enableAttrib(
-    this.curStrokeShader.attributes.aDirection.location,
-    4,
-    gl.FLOAT,
-    false,
-    0,
-    0
-  );
-
-  // allocate space for vertex positions
-  this._bindBuffer(
-    this.gHash[gId].vertexBuffer,
-    gl.ARRAY_BUFFER,
-    this._vToNArray(obj.vertices),
-    Float32Array,
-    gl.STATIC_DRAW
-  );
-
-  this.curFillShader.enableAttrib(
-    this.curFillShader.attributes.aPosition.location,
-    3,
-    gl.FLOAT,
-    false,
-    0,
-    0
-  );
-
-  // allocate space for faces
-  this._bindBuffer(
-    this.gHash[gId].indexBuffer,
-    gl.ELEMENT_ARRAY_BUFFER,
-    this._flatten(obj.faces),
-    Uint16Array,
-    gl.STATIC_DRAW
-  );
+  var geometry = this.gHash[gId];
 
-  // allocate space for normals
-  this._bindBuffer(
-    this.gHash[gId].normalBuffer,
-    gl.ARRAY_BUFFER,
-    this._vToNArray(obj.vertexNormals),
-    Float32Array,
-    gl.STATIC_DRAW
-  );
-
-  this.curFillShader.enableAttrib(
-    this.curFillShader.attributes.aNormal.location,
-    3,
-    gl.FLOAT,
-    false,
-    0,
-    0
-  );
+  geometry.numberOfItems = obj.faces.length * 3;
+  geometry.lineVertexCount = obj.lineVertices.length;
 
-  // tex coords
-  this._bindBuffer(
-    this.gHash[gId].uvBuffer,
-    gl.ARRAY_BUFFER,
-    this._flatten(obj.uvs),
-    Float32Array,
-    gl.STATIC_DRAW
-  );
+  this._useColorShader();
 
-  this.curFillShader.enableAttrib(
-    this.curFillShader.attributes.aTexCoord.location,
-    2,
-    gl.FLOAT,
-    false,
-    0,
-    0
-  );
-  //}
-};
+  if (this.curStrokeShader.attributes.aPosition) {
```

Comment:
this looks great! it occurs to me that now it's probably worth adding a comment about what each if-block is doing here to make the code a little more scannable. we probably should have had this already. :)